src/actions/tabs.js

```diff
@@ -106,6 +106,7 @@ export function closeTabsForMissingThreads(workers: Worker[]) {
       if (sourceId) {
         const source = getSourceFromId(getState(), sourceId);
         if (
+          source &&
```

Comment:
I think we should use getSource instead of getSourceFromId. This way we're telling flow source can be null.

Reply:
Sure, just pushed a fix for this.  Having these two methods with identical implementations and different return types is kind of silly though, I'll try to fix this as part of bug 1518884.

Reply:
agreed... `getSourceFromId` was a bad idea :/